infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -374,4 +464,42 @@ public void shutdown() {
     public void enforceSnapshotSync(DiscoveryServiceEvent event) {
         replicationManager.enforceSnapshotSync(event);
     }
+
+    /**
+     * Check if the local cluster is a connection receiver for any of the discovered sessions
+     *
+     * @return true if the local cluster is a connection receiver for any session
+     */
+    public boolean isConnectionReceiver() {
+        Set<String> connectionEndpoints = topology.getRemoteClusterEndpoints().keySet();
+        for(LogReplicationSession session : sessions) {
+            if (!connectionEndpoints.contains(session.getSinkClusterId()) &&
```

Comment:
I got stuck reading this method and its implications because if we apply Demorgan's law we can simplify and say if remoteEndpoint is in session's (source | sink )-> return false, but is that sufficient to not start the LogReplcationServerNode ?

Reply:
Sure, if it simplifies. this would now be an if-else statement, because we need to continue checking until we get a session for which the remote will be connecting to the local cluster or we exhaust all the sessions found.

I think the check is sufficient:
1. the in-memory sessions only have the session for which the local cluster is either a source or a sink
2. LogReplcationServerNode is only needed when there exists a session to which the local cluster does not connect...implying that the local cluster will receive a connection.

This works when the local cluster has the ability to know of all possible sessions. 
If we have a case where this isn't possible, then we'll have a problem where the remote GRPC client may try to connect to the local but the local cluster hasn't started the GRPC server..
For now this case is not possible...